### ISSUE TITLE
Create tracklets does not allow only one individual (in multi-animal code 2.2rc1)

### ISSUE THREAD

**Issue by kevinbolding** [DeepLabCut/DeepLabCut#1215]
**Your Operating system and DeepLabCut version**

Please state your operating system, env, and which version of DeepLabCut you are using.
Example: Ubuntu 16.04 LTS, with an Anaconda Env, & DeepLabCut1.x or 2.x.
Windows Server 2019 Standard, first installed with Anaconda Env, then dl'ed git and set up in that environment, DeepLabCut2.2rc1, manually ran each line from reinstall.sh

Please complete the following information about your system:

OS: Windows Server 2019 Standard
DeepLabCut Version: 2.2rc1
Anaconda env used: DLC-GPU

**Describe the problem**

The proximate issue is that the Refine tracklets step appears to be ignoring any attempt I make to track only one animal. This most clearly manifests in two ways:
-  Though I have specified a single individual named mouse1 in the config.yaml , the Refine Tracklets GUI has a legend for ind1 and ind2. I have not found any combination of pcutoff, max_age, or max gap size or any other variable that will get me back down to a single mouse, much less a single mouse labeled 'mouse1'.
- At first the "Number of Animals" is set to 1. However, if I interact at all with the "Number of Animals" box in the Refine tracklets window, the number sets itself to 2 and will not allow to go any lower.

Please place code inside this:

<details><p>

Nothing special happens in the terminal during this issue.

</p></details>

**How to Reproduce the problem**
1. Convert single animal project made in DeepLabCut2.1 to multi-animal project, carefully following instructions in convert_maDLC.md
2. Crop images and labels, create multianimal training dataset, train overnight. I ran 200000 iterations because I had to reduce batch size because of my GPU.
3. Use GUI to analyze a test video and convert to tracklets with ellipse
4. Get to Refine tracklets step. Here is where I encountered issues. 

**Screenshots**
![Screen Shot 2021-05-03 at 10 07 31 PM](https://user-images.githubusercontent.com/9322812/116953919-81d32800-ac5c-11eb-9c12-43b502f1f4e2.png)

**Additional context**
The problem I described above is the proximate issue and may be a legitimate bug. 

The ultimate issue is that I cannot understand how to get a continuous single-animal location prediction for all frames. I have been searching the gitter and issues extensively. I believe my issue may have similarities with issue #949 or issue #827.

For context, evaluating the model produces good values: 
Average Euclidean distance to GT per individual (in pixels)
individuals
mouse1    2.448599
single    2.222496

The video created with all detections after analysis looks good to me except there are extraneous labels outside the arena. In particular, the tailbase is nearly constantly labeled. This should be visible and easy for the tracklets to follow, although this is really where my lack of understanding about how pose estimates turn into tracklets really shows.

My skeleton is overcomplete but not exhaustive. I see this as a first suggestion on many issues. One thing that is unclear about this is whether you really mean literally that every part should be connected to every other part. If that is so, why not automatically/programmatically do so instead of having user input? I am attaching a screenshot of my skeleton below.

In any case, I'm hoping that tugging on this thread about Refine Tracklets forcing me to have 2 individuals will point me toward the guidance to get back to usable output, which for me is simply a list of xy coordinates with likelihoods for every frame. Since I recently updated to 2.2rc1 I will be trying out a training session with an updated single animal model overnight, though I had understood in my reading that the multi-animal version may serve best even for single animal conditions.
![Screen Shot 2021-05-03 at 10 06 55 PM](https://user-images.githubusercontent.com/9322812/116953910-767ffc80-ac5c-11eb-92c1-321c1e5e04f5.png)

**Comment by kevinbolding** [comment]
In the process of reaching this stage of the analysis, I discovered a couple of other things that might be true bugs.
1. When I converted to maDLC project and created a multianimal training set, the iteration number did not increase. So new checkpoints were written into a previously occupied training folder. This caused a problem later because during analysis the previous checkpoints (which went up to 730000 iterations) were loaded rather than the last multianimal checkpoint.
2. It appears that when I used cropimagesandlabels, the first video in my config file was processed correctly, but there was an error in the accompanying config file changes. The first video entry in video_sets wasn't changed to a cropped suffix and the cropping parameters weren't changed. I changed this manually. 
I got around these, but they may be worth a second look to see if things are working as expected during those steps.

**Comment by AlexEMG** [comment]
Just re: 1.: this is intended. The counter always starts from 0, when training. If you do not want the models/snapshots to be in the same folder (and possibly overwrite each other), just update the iteration variable. See Nature Protocols: Stage X continued: (optional) refinement of labels—augmentation of the training dataset https://www.nature.com/articles/s41596-019-0176-0

**Comment by MMathisLab** [comment]
we now describe 2-14 animals in the paper, and this is what it's optimized for. I agree we can add a feature to use a single animal only again, but really using EfficientNets for single animal is truly smarter; it's computationally more complex to do multi-animal, and as we describe in the paper, the only advantage is when animals are very close. 

From the blog:

🥇 **Is multi-animal truly good for your problem?** We saw many animals that were easily visually distinguishable being used with #maDLC beta, which by default has the strong assumption there should be no unique visual properties. Even two mice with one having a marked tail violates this assumption, unless you use our ID network output (see next point).

> The ultimate issue is that I cannot understand how to get a continuous single-animal location prediction for all frames. I have been searching the gitter and issues extensively. I believe my issue may have similarities with issue #949 or issue #827.

This is completely different from beta to now; so it's really not relevant. 

> The video created with all detections after analysis looks good to me except there are extraneous labels outside the arena. In particular, the tailbase is nearly constantly labeled. This should be visible and easy for the tracklets to follow, although this is really where my lack of understanding about how pose estimates turn into tracklets really shows.

May I ask, did you read the paper? All the methods are fully described. 

> My skeleton is overcomplete but not exhaustive. I see this as a first suggestion on many issues. One thing that is unclear about this is whether you really mean literally that every part should be connected to every other part. If that is so, why not automatically/programmatically do so instead of having user input? I am attaching a screenshot of my skeleton below.

Please read the new docs, and the paper, and the blog clearly described this is no longer needed, btw. Here is an excerpt from the blog. And please we worked really hard on this, and the little we ask is for a doc check ;), then we are happy to help. 

BLOG: http://www.mackenziemathislab.org/deeplabcutblog/2021/4/2/multi-animal-deeplabcut-the-full-22-release

PAPER: https://www.biorxiv.org/content/10.1101/2021.04.30.442096v1.full.pdf

🗝 **The skeleton, you need(ed) to over-connect…** And when we mean over-connect, we really meant everything to everything is best. Now, we (1) take this decision away from the user, and (2) introduce a novel fully-data driven approach. Using your data, we now find the ultimate skeleton for your data to get the best performance with the best speed. Super cool, right?

**Comment by kevinbolding** [comment]
It appears that I jumped the gun creating an issue here.  I had been monitoring https://github.com/DeepLabCut/DeepLabCut for updates when I learned that 2.2rc1 was imminent, and there was some lag between finding the new code and my becoming aware of the preprint and blog. I had attempted to do due diligence and make a good faith attempt at solving my own issues by searching for similar problems to my own in the GitHub issues history, but I now see those refer to deprecated code.

I have not read the preprint yet. I will do so before asking for any more assistance. It sounds in any case that the single-animal, conventional DLC remains more appropriate for my use case.

Please feel free to close or delete this issue.

**Comment by MMathisLab** [comment]
Happy to leave it open and add a single animal option, but just wanted to ask to meet us half way on this one.

**Comment by kevinbolding** [comment]
FWIW I think the following statements from[ this part of the docs](https://github.com/DeepLabCut/DeepLabCut/blob/master/docs/UseOverviewGuide.md) are what led me to think the multi-animal model was the correct choice. Advanced tracking features and updated network capabilities are altogether too enticing to resist.

> 💜 DLC 2.2: as of 2.2 we support "multi-animal projects," but these new features can also be used on single animals too (details below). The workflow is highly similar, but with a few key additional steps. Please carefully review the functions below for more details. You can search/look for maDeepLabCut for specific steps that are changed, or see this more comprehensive guide here

> What scenario do you have?
> I have single animal videos, but I want to use the advanced tracking features & updated network capabilities introduced (for multi-animal projects) in DLC2.2:
> quick start: when you create_new_project just set the flag multianimal=True.
